sprockets/mixins/http/__init__.py

```diff
@@ -103,31 +302,40 @@ def __init__(self, *args, **kwargs):
         :param str user_agent: The str used for the ``User-Agent`` header,
             default used if unspecified.
         :param bool validate_cert: For HTTPS requests, validate the server's
-            certifacte? Default is True
+            certificate? Default is True
         :param bool allow_nonstandard_methods: Allow methods that don't adhere
             to the HTTP spec.
+        :param set dont_retry: A list of status codes that will not be retried
+            if an error is returned. Default: set({})
         :rtype: HTTPResponse
 
         """
+        response = HTTPResponse()
```

Comment:
You're setting the start time upon creation so it's probably better to create this as close to the first request as possible. Especially with *request* serialization currently occurring after this right now.

Reply:
I'm explicitly starting it as close to the start of the method as possible to get the total timing in place.